src/multitask_personalization/envs/pybullet/pybullet_csp.py

```diff
@@ -49,141 +51,107 @@
-        # NOTE: separate out "retract" here; see note in _get_pick_plan().
-        self._sim.set_state(obs)
-        held_obj = obs.held_object
-        assert held_obj is not None
-        held_obj_id = self._sim.get_object_id_from_name(held_obj)
-        surface_names = sorted(self._sim.get_surface_names())
-        # In the near future, we will figure out how best to add placement pose
-        # as another variable in the CSP, but only when the user is holding a
-        # book that is not the one that was asked for. This is part of a bigger
-        # thing which is figuring out how to make CSP generation conditional on
-        # the current state (in a nice implementation kind of way).
-        for _ in range(100000):
-            # Sample a surface.
-            surface_name = surface_names[self._rng.choice(len(surface_names))]
-            surface_id = self._sim.get_object_id_from_name(surface_name)
-            surface_link_ids = sorted(self._sim.get_surface_link_ids(surface_id))
-            # Sample a surface link.
-            surface_link_id = self._rng.choice(surface_link_ids)
-            placement_pose = next(
-                generate_surface_placements(
-                    held_obj_id,
-                    surface_id,
-                    self._rng,
-                    self._sim.physics_client_id,
-                    surface_link_id=surface_link_id,
-                )
+        if obs.held_object is None:
+            # Pick up the target book.
+            return get_plan_to_pick_object(
+                obs,
+                book_description,
+                book_grasp,
+                self._sim,
+                max_motion_planning_candidates=self._max_motion_planning_candidates,
             )
-            self._sim.set_state(obs)
-            plan = get_plan_to_place_object(
+        if obs.held_object == book_description:
+            # Handover the book.
+            plan = get_plan_to_handover_object(
                 obs,
-                held_obj,
-                surface_name,
-                placement_pose,
+                book_description,
+                handover_pose,
                 self._sim,
-                max_motion_planning_time=1.0,
+                self._seed,
                 max_motion_planning_candidates=self._max_motion_planning_candidates,
-                surface_link_id=surface_link_id,
             )
-            if plan is not None:
-                return plan
-        raise RuntimeError("Could not find placement plan")
-
-    def _get_mission_from_obs(self, obs: PyBulletState) -> str | None:
-        if obs.human_text is None:
-            return None
-        # Hardcode rules to save some LLM costs for now.
-        if "Please bring me a book to read" in obs.human_text:
-            return "hand over book"
-        if "Put away the thing you're holding" in obs.human_text:
-            return "put away held object"
-        return None
+            plan.append((2, None))
```

Comment:
yeah that would be good. I found another possible pattern when starting the cooking env. we could swap that in here as well. since this isn't a new change for this PR, I'll leave it until later